vowpalwabbit/example_predict.cc

```diff
@@ -4,18 +4,48 @@
 
 #include "example_predict.h"
 
-safe_example_predict::safe_example_predict()
+example_predict::iterator::iterator(features* feature_space, namespace_index* index)
```

Comment:
whats the difference between `example_predict` and `safe_example_predict`?

Reply:
nit: We should add a `using iterator = T*` to v_array. Then you can change `namespace_index*` to `v_array<T>::iterator` to make it easier to swap out with other standard container types in the future. Really just a minor nit though

Reply:
`safe_example_predict` is just `example_predict` with a destructor that properly releases resources. Now that `example_predict` does this `safe_example_predict` is now marked deprecated.